integration/test_collection.py

```diff
@@ -1464,7 +1471,11 @@ class Data(TypedDict):
     for _ in range(3):
         # get the property and sort them - order returned by weaviate is not identical to the order inserted
         all_data: list[int] = sorted(
-            [int(obj.properties["data"]) for obj in collection.iterator(Data)]
+            [
+                int(obj.properties["data"])
+                for obj in collection.iterator(return_metadata=metadata, return_properties=Data)
+                if metadata is None or obj.metadata.creation_time_unix is not None
```

Comment:
why the if?

Reply:
For the two parametrized cases, I'll add them as parameters to the tests to make it clearer!

Reply:
but it evaluates to true for both, so we could just remove it?

Reply:
It enforces that `obj.metadata.creation_time_unix` is not `None` when `metadata=Metadata(creation_time_unix=True)` in the parameter since the list will be empty if it is `None` and thereby fail the assertion. Pretty convoluted so will make it clearer!